### PR TITLE
chore: make updates to toc.yml before upload

### PR DESCRIPTION
**Pull request by eaball35**
- adding README as overview
- sorting toc on package
- adding product hierarchy
- script cleanup

### CODE REVIEW

**Comment by eaball35** [comment]
Tested these changes out in single library (on branch in java-memcache) and is working correctly as expected

**Comment by eaball35** [comment]
b/181142127
b/179920619
b/179799230

File: synthtool/gcp/templates/java_library/.kokoro/release/prepare_docs.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2021 Google Inc.
```
Comment (chingor13) [inline synthtool/gcp/templates/java_library/.kokoro/release/prepare_docs.py:1]:
```suggestion
# Copyright 2021 Google LLC
```

**Review by chingor13 — CHANGES_REQUESTED** [review]
Is this testable anywhere? Ideally, we can test this script (perhaps make this part of synthtool)

**Review by chingor13 — CHANGES_REQUESTED** [review]
The CI failure is because the python template is being compiled in-place, then the template engine is confused because it sees a new compiled.py file and tries to load it as a utf-8 file (which it's not).

From an offline chat, we decided the python script belongs in a single place in a python library (gcp-docuploader/publisher perhaps?) rather than as a standalone python file in 100+ repos.

**Comment by eaball35** [comment]
Moving this out of synthtool and into docuploader instead https://github.com/googleapis/docuploader/pull/58

**Pull request closed without merging**